### ISSUE TITLE
Improve pk_wrap calling convention and check for remaning ECP dependencies in PK files

### ISSUE THREAD

**Issue by valeriosetti** [Mbed-TLS/mbedtls#7460]
Following the [comments](https://github.com/Mbed-TLS/mbedtls/pull/7202#issuecomment-1508063453) in #7073 and discussion on slack, here is the to do list to be addressed before starting #7073 and #7074:

- [ ] Add missing USE_PSA_INIT()/DONE() in tests - actually while at it we could do this in a systematic way: according to [our own documentation](https://github.com/Mbed-TLS/mbedtls/blob/development/include/mbedtls/mbedtls_config.h#L1933) all PK, X.509 and SSL tests should have it (unless they already have MD_OR_USE_PSA_INIT()/DONE() of course, or a plain PSA_INIT()/PSA_DONE() for 1.3 tests). (Note: there's one in mbedtls_test_ssl_perform_handshake(), so test functions that are merely a wrapper around that don't need it.)
    - #7461 
- [ ] Change the calling convention to pass the full pk_context for existing wrappers - might be worth checking that the CI fully passes at this point (actually those two commits could even be a separate PR). See also [this comment](https://github.com/Mbed-TLS/mbedtls/pull/7202#discussion_r1170999430)
    - https://github.com/Mbed-TLS/mbedtls/pull/7463
- [x] grep for remaining mbedtls_ecp functions in PK files and think about what's the plan for replacing/removing each one of them
    - https://github.com/Mbed-TLS/mbedtls/pull/7465

## Note for reviewers/gatekeepers
All the points mentioned above will be addressed by separate PRs, so please wait for all of them to be addressed before closing this issue.

**Comment by tom-cosgrove-arm** [comment]
@valeriosetti Any chance of a title that gives some idea as to what the issue is about, please, so that everyone doesn't have to read the description (or two other issues) before they know if they need to look at this further or not?

**Comment by mpg** [comment]
> grep for remaining mbedtls_ecp functions in PK files

Actually, on second thought, grepping the source is not going to be very helpful, as it will return a lot of calls from the `!defined(MBEDTLS_USE_PSA_CRYPTO)` path. Perhaps what we want to do instead is build with `MBEDTLS_USE_PSA_CRYPTO` and look for external dependencies on `mbedtls_ecp_` functions in the object file, similarly to what `docs/architecture/psa-migration/syms.sh` instead we'd do it with `pk.o`, `pkwrite.o`, `pkparse.o` instead of `libmbedx509.a` and `libmbedtls.a`.

**Comment by gilles-peskine-arm** [comment]
> grepping the source is not going to be very helpful, as it will return a lot of calls from the !defined(MBEDTLS_USE_PSA_CRYPTO) path

Run through cpp or unifdef?

**Comment by mpg** [comment]
I remember trying cpp, and IIRC the problem is we get many false positives from `#include` directives. I remember considering `unifdef` but I don't remember why I didn't end up using it and finally found it more convenient to just look for unresolved symbols in the object files.

**Comment by AndrzejKurek** [comment]
I believe that only one of three points has been addressed so far.

**Comment by valeriosetti** [comment]
> I believe that only one of three points has been addressed so far.

Yes, I didn't noticed this! Thanks!
Perhaps it's my fault that I already ticked the check-boxes after PRs' creation. Now I will select only the merged ones, to avoid confusion ;)

**Comment by AndrzejKurek** [comment]
> > I believe that only one of three points has been addressed so far.
> 
> Yes, I didn't notice this! Thanks! Perhaps it's my fault that I already ticked the check-boxes after PRs' creation. Now I will select only the merged ones, to avoid confusion ;)

I think that the problem was writing `resolves xxx` in the merged PR :) Even though there was "partially" before "resolves", github doesn't understand that and picks it up as a resolution for the issue :)
